src/components/ProductList/ProductList.vue

```diff
@@ -0,0 +1,136 @@
+<template>
+
+  <ul class="items-container"> 
+    <ProductListItem v-for="item in items" 
+    :id="item.id" 
+    :key="item.id"
+    :title="item.title" 
+    :url="item.url"
+    :price="item.price"
+     @delete-item="openDeleteWindow"
+     @editing-item="openEditWindow"
+    />
+      <Modal 
+     v-if="showModal"
+    :initialTitle="item.title"
+    :initialPrice="item.price"
+    :initialUrl="item.url"
+    :itemId="item.id"
+    :title="action"
+    @close-modal="handleClose"
+    >
+    <div slot="header">
+      <p>{{action}}</p>
+    </div>
+    <div  slot="modal-body">
+      <template v-if="action === 'Deleting'">
+        <p>Deleted item will be lost!</p>
+      </template>
+      <template v-else>
+       <add-edit-form 
+       :action="action" 
+       :item="item"
+       @close-modal="handleClose"
+       @save-edit="saveEditedItem"
+       ></add-edit-form>
+      </template>
+    </div>
+    <div slot="modal-footer"> 
+       <template v-if="action === 'Deleting'">
+         <VButton @click="handleClose" text="Cancel"/>
+        <VButton class="button-delete"  @click="handleDelete" text="Delete"/>
+      </template>
+    </div>
+    </Modal>
+      
+  </ul>
+
+ 
+</template>
+
+<script>
+import Modal from "../Modal/Modal.vue"
+import ProductListItem from "../ProductListItem/ProductListItem.vue"
+import VButton from "../VButton/VButton.vue"
+import AddEditForm from "../AddEditForm/AddEditForm.vue"
+import {deleteData} from "../../api/apiCalls"
+
+export default {
+  components: { ProductListItem, Modal, VButton, "add-edit-form": AddEditForm },
+  data() {
+    return {
+      items: [],
+      showModal: false,
+      action: null,
+      itemToDeleteId: null,
+      itemToEditId: null,
+      item: {},
+    }
+  },
+  methods: {
+ 
+    async handleDelete () {
+      const oldDtata = [...this.items]
```

Comment:
Is there a reason you are creating a copy of this array?  
It looks like it would be enough to just simply
```js
this.items = this.items.filter(item => item.id !== this.itemToDeleteId)
```
when needed. 🤔